springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/payload/NamedSchemaObject.java

```diff
@@ -1,16 +1,25 @@
 // SPDX-License-Identifier: Apache-2.0
 package io.github.springwolf.core.asyncapi.scanners.common.payload;
 
+import io.github.springwolf.asyncapi.v3.model.channel.message.MessageReference;
 import io.github.springwolf.asyncapi.v3.model.schema.SchemaObject;
 import jakarta.annotation.Nullable;
 
 /**
  * Encapsulates the resolved name for the contained schema.
  * @param name The fully qualified name or the simple name of the schema.
  * @param schema The SchemaObject.
+ * @param schemaPayload The schema-payload to be inserted in the message, when not null this schema will override the payload of the message.
```

Comment:
nitpicking: I wonder if this is rather something like a `composedSchema` (composed = anyOf, allOf, oneOf; which right is only used for polymorphic payloads)

Currently the naming represents what it should be used for vs. what it actually is/represents

Reply:
Actually, I just saw that it is called polymorphism in OpenAPI: https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/
ComposedSchema is used by Swagger, however SchemaObjects are already Springwolf mapped AsyncAPI schemas.
tldr: I am still unsure about the typical CS problem called 'naming things'